### ISSUE TITLE
Trouble replicating experiments

### ISSUE THREAD

**Issue by NicholaiStaalung** [pycaret/pycaret#505]
Hello,

When running setup() and either create_model() or compare_models() i experience that results vary slightly on each new experiment.

In the documentation it says that session_id is suppose to control this but thats not what im experiencing. I have looked through the code and i do see that session_id gets passed to seed, which gets passed on to any random_state argument when necessary.

What am i missing? Some randomness that goes on in the background? Anyone else experiencing this?

**Comment by NicholaiStaalung** [comment]
> @NicholaiStaalung Few other people have also reported this. I am unable to find the reason yet why this happens.
> 
> Would you share the detail of your OS including the build number as well as the detail of Python you are using and the environment?
> 
> When you run your experiment, PyCaret generates a `logs.log` file in your local directory. Look for the `setup` function initialization and at the beginning of that function, all the details related to your OS are captured. Can you post that here?
> 
> Maybe we will get some clue if it is behaving like this on a specific OS or build. I will also leave this ticket open for any expert in this area to comment.

Im on Ubuntu 20.04

2020-08-07 11:54:59,313:INFO:PyCaret Regression Module
2020-08-07 11:54:59,314:INFO:version 2.0
2020-08-07 11:54:59,314:INFO:Initializing setup()
2020-08-07 11:54:59,314:INFO:USI: fb2f
2020-08-07 11:54:59,315:INFO:setup(data=(524, 44), target=Score, train_size=0.7, sampling=True, sample_estimator=None, categorical_features=None, categorical_imputation=constant, ordinal_features=None,
                    high_cardinality_features=None, high_cardinality_method=frequency, numeric_features=None, numeric_imputation=mean, date_features=None, ignore_features=None, normalize=False,
                    normalize_method=zscore, transformation=False, transformation_method=yeo-johnson, handle_unknown_categorical=True, unknown_categorical_method=least_frequent, pca=False, pca_method=linear,
                    pca_components=None, ignore_low_variance=False, combine_rare_levels=False, rare_level_threshold=0.1, bin_numeric_features=None, remove_outliers=False, outliers_threshold=0.05,
                    remove_multicollinearity=False, multicollinearity_threshold=0.9, remove_perfect_collinearity=False, create_clusters=False, cluster_iter=20,
                    polynomial_features=False, polynomial_degree=2, trigonometry_features=False, polynomial_threshold=0.1, group_features=None,
                    group_names=None, feature_selection=False, feature_selection_threshold=0.8, feature_interaction=False, feature_ratio=False, interaction_threshold=0.01, transform_target=False,
                    transform_target_method=box-cox, data_split_shuffle=True, folds_shuffle=False, n_jobs=-1, html=True, session_id=123, log_experiment=False,
                    experiment_name=None, log_plots=False, log_profile=False, log_data=False, silent=False, verbose=True, profile=False)
2020-08-07 11:54:59,315:INFO:Checking environment
2020-08-07 11:54:59,315:INFO:python_version: 3.8.2
2020-08-07 11:54:59,315:INFO:python_build: ('default', 'Jul 16 2020 14:00:26')
2020-08-07 11:54:59,315:INFO:machine: x86_64
2020-08-07 11:54:59,331:INFO:platform: Linux-5.4.0-42-generic-x86_64-with-glibc2.29
2020-08-07 11:54:59,332:WARNING:cannot find psutil installation. memory not traceable. Install psutil using pip to enable memory logging. 
2020-08-07 11:54:59,332:INFO:Checking libraries
2020-08-07 11:54:59,332:INFO:pd==1.1.0
2020-08-07 11:54:59,333:INFO:numpy==1.18.5
2020-08-07 11:54:59,333:INFO:sklearn==0.23.2
2020-08-07 11:54:59,343:INFO:xgboost==1.1.1
2020-08-07 11:54:59,352:INFO:lightgbm==2.3.1
2020-08-07 11:54:59,366:INFO:catboost==0.24
2020-08-07 11:54:59,650:INFO:mlflow==1.10.0
2020-08-07 11:54:59,650:INFO:Checking Exceptions
2020-08-07 11:54:59,650:INFO:Preloading libraries
2020-08-07 11:54:59,650:INFO:Preparing display monitor
2020-08-07 11:54:59,658:INFO:Importing libraries
2020-08-07 11:54:59,774:INFO:Copying data for preprocessing
2020-08-07 11:54:59,775:INFO:Declaring global variables
2020-08-07 11:54:59,780:INFO:Declaring preprocessing parameters
2020-08-07 11:54:59,780:INFO:Importing preprocessing module
2020-08-07 11:55:00,169:INFO:Creating preprocessing pipeline
2020-08-07 11:55:00,227:INFO:Preprocessing pipeline created successfully
2020-08-07 11:55:00,227:INFO:No inverse transformer found
2020-08-07 11:55:00,227:INFO:Creating grid variables
2020-08-07 11:55:00,229:INFO:Creating global containers
2020-08-07 11:55:00,291:INFO:create_model_container: 0
2020-08-07 11:55:00,291:INFO:master_model_container: 0
2020-08-07 11:55:00,291:INFO:display_container: 0
2020-08-07 11:55:00,291:INFO:setup() succesfully completed......................................
2020-08-07 11:55:00,292:INFO:Initializing compare_models()
2020-08-07 11:55:00,292:INFO:compare_models(blacklist=None, whitelist=None, fold=10, round=4, sort=R2, n_select=1, turbo=True, verbose=True)
2020-08-07 11:55:00,292:INFO:Checking exceptions
2020-08-07 11:55:00,292:INFO:Preloading libraries
2020-08-07 11:55:00,292:INFO:Preparing display monitor

**Comment by pycaret** [comment]
@NicholaiStaalung Thanks

**Comment by pycaret** [comment]
@NicholaiStaalung 

This issue has been fixed in the `pycaret-nightly` release. Please install pycaret-nightly using `pip install pycaret-nightly`. Feel free to open a new issue, in case the problem persists. This fix will be published to PyPi as `pycaret==2.2.0` on October 28, 2020.

**Comment by NicholaiStaalung** [comment]
Thanks for getting back to me! Love your work
